File: LSP+ISP/Solution/MarsRoverKata/Commands/ForwardCommand.cs
namespace MarsRoverKata.Commands
{
    public class ForwardCommand : ICommand
    {
        private Rover Rover { get; set; }

        public ForwardCommand(Rover rover)
        {
            Rover = rover;
        }

        public bool Execute()
        {
            return Rover.MoveForward();
        }
    }
}

File: LSP+ISP/Java_Exercise/MarsRoverKata/src/main/java/MarsRoverKata/Commands/ForwardCommand.java
package MarsRoverKata.Commands;

import MarsRoverKata.Rover;

public class ForwardCommand implements ICommand {
    private Rover rover;

    public Rover getRover() {
        return rover;
    }

    public void setRover(Rover rover) {
        this.rover = rover;
    }

    public ForwardCommand(Rover rover) {
        this.rover = rover;
    }

    public boolean execute() {
        return rover.moveForward();
    }
}

